include/zenohcxx/api.hxx

```diff
@@ -670,6 +670,84 @@ struct Timestamp : Copyable<::z_timestamp_t> {
+struct AttachmentView : public Copyable<::z_attachment_t> {
+    using Copyable::Copyable;
+
+    /// The body of a loop over an attachment's key-value pairs.
+    ///
+    /// `key` and `value` are loaned to the body for the duration of a single call.
+    ///
+    /// Returning `true` is treated as `continue`.
+    /// Returning `false` is treated as `break`.
+    typedef std::function<bool(const BytesView& key, const BytesView& value)> IterBody;
+
+    /// The driver of a loop over an attachment's key-value pairs.
+    ///
+    /// This function is expected to call `body` once for each key-value pair.
+    /// Returning `false` value immediately (breaking iteration).
+    typedef std::function<bool(const IterBody& body)> IterDriver;
+
+    /// @name Constructors
+
+    /// @brief AttachmentView constructor by interation driver
+    AttachmentView(const AttachmentView::IterDriver& _iter_driver)
+        : Copyable({static_cast<const void*>(&_iter_driver),
+                    [](const void* data, z_attachment_iter_body_t body, void* ctx) -> int8_t {
+                        const IterDriver* _iter_driver = static_cast<const IterDriver*>(data);
+                        return (*_iter_driver)([&body, &ctx](const BytesView& key, const BytesView& value) {
+                            return body(key, value, ctx);
+                        });
+                    }}) {}
+
+    /// @brief AttachmentView constructor by the container which allows
+    /// iterate by std::pair<std::string_view, std::string_view>
+    /// (e.g. std::map<std::string_view, std::string_view>).
+    template <typename T>
+    AttachmentView(const T& pair_container)
+        : Copyable({static_cast<const void*>(&pair_container),
+                    [](const void* data, z_attachment_iter_body_t body, void* ctx) -> int8_t {
+                        const T* pair_container = static_cast<const T*>(data);
+                        for (const auto& it : *pair_container) {
+                            int8_t ret = body(BytesView(it.first), BytesView(it.second), ctx);
+                            if (ret) {
+                                return ret;
+                            }
+                        }
+                        return 0;
+                    }}) {}
+
+    /// @name Methods
+
+    /// Returns the item value from the attachment by key
+    /// @return the item value
+    BytesView get(const BytesView& key) const { return ::z_attachment_get(*this, key); }
+
+    /// Checks if the attachment is initialized
+    /// @return true if the attachment is initialized
+    bool check() const { return ::z_attachment_check(this); }
+
+    // Iterate over attachment's key-value pairs, breaking if `body` returns a non-zero
+    // value for a key-value pair, and returning the latest return value.
+    // `context` is passed to `body` to allow stateful closures.
+    // This function takes no ownership whatsoever.
+    /// @return the latest return value
```

Comment:
Probably should return void now - it's always returns false. Maybe later the variant with "ErrNo& error" with real int value from C will be added, like in all other functions which separates bool and C numeric error